pyro/util.py

```diff
@@ -52,7 +52,7 @@ def warn_if_nan(value, msg=""):
     also works with numbers.
     """
     if torch.is_tensor(value) and value.requires_grad:
-        value.register_hook(lambda x: warn_if_nan(x, msg))
+        value.register_hook(lambda x: warn_if_nan(x, "grad " + msg))
```

Comment:
I find this to be useful when checking which grad is problematic (e.g. warnings for 2nd derivative say "grad grad myvar")

Reply:
Feel free to submit this as a 1-line PR :smile: